util/cpt_upgraders/riscv-hext.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) 2024 National and Kapodistrian University of Athens
+# All rights reserved.
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+def upgrader(cpt):
+    """
+    Update the checkpoint to support RVH implemtation.
+    """
+    import re
+
+    for sec in cpt.sections():
+        # # Upgrade interrupts to add HVIP reg
+        res = re.search(r"(.*processor.*\.core.*)\.interrupts$", sec)
+        if res and cpt.get(res.groups()[0] + ".isa", "isaName") == "riscv":
+            # Only update for RISCV interrupts
+            # Add hvip_ulong dummy value
+            cpt.set(sec, "hvip_ulong", "0")
+
+        res = re.search(r"(.*processor.*\.core.*)\.xc.*", sec)
+        if res and cpt.get(res.groups()[0] + ".isa", "isaName") == "riscv":
+            # Only update for RISCV XCs
+            # Add hvip_ulong dummy value
+            cpt.set(sec, "hvip_ulong", "0")
+
+        # Update the miscregs
+        if (
+            re.search(r".*processor.*\.core.*\.isa$", sec)
+            and cpt.get(sec, "isaName") == "riscv"
+        ):
+            # Updating RVH misc registers (dummy values)
+            mr = cpt.get(sec, "miscRegFile").split()
+            if len(mr) == 187:
+                print(
+                    "MISCREG_* RVH registers already seem " "to be inserted."
+                )
+            else:
```

Comment:
Do we need to be worried that we change the name from MISCREG_FFLAGS to HELPER_...?